engines/sherlock/objects.cpp

```diff
@@ -0,0 +1,1131 @@
+						return;
+
+					case SOLID:
+					case TALK:
+						// Stop walking
+						if (obj._aType == TALK) {
+							obj.setFlagsAndToggles();
+							talk.talkTo(obj._use[0]._target);
+						} else {
+							people.gotoStand(*this);
+						}
+						break;
+
+					case TALK_EVERY:
+						if (obj._aType == TALK_EVERY) {
+							obj._type = HIDDEN;
+							obj.setFlagsAndToggles();
+							talk.talkTo(obj._use[0]._target);
+						} else {
+							people.gotoStand(*this);
+						}
+						break;
+
+					case FLAG_SET:
+						obj.setFlagsAndToggles();
+						obj._type = HIDDEN;
+						break;
+
+					case WALK_AROUND:
+						if (objBounds.contains(people._walkTo.front())) {
+							// Reached zone
+							people.gotoStand(*this);
+						} else {
+							// Destination not within box, walk to best corner
+							Common::Point walkPos;
+
+							if (spritePt.x >= objBounds.left && spritePt.x < objBounds.right) {
+								// Impact occurred due to vertical movement. Determine whether to
+								// travel to the left or right side
+								if (_delta.x > 0)
+									// Go to right side
+									walkPos.x = objBounds.right + CLEAR_DIST_X;
+								else if (_delta.x < 0) {
+									// Go to left side
+									walkPos.x = objBounds.left - CLEAR_DIST_X;
+								} else {
+									// Going straight up or down. So choose best side
+									if (spritePt.x >= (objBounds.left + objBounds.width() / 2))
+										walkPos.x = objBounds.right + CLEAR_DIST_X;
+									else
+										walkPos.x = objBounds.left - CLEAR_DIST_X;
+								}
+
+								walkPos.y = (_delta.y >= 0) ? objBounds.top - CLEAR_DIST_Y :
+									objBounds.bottom + CLEAR_DIST_Y;
+							} else {
+								// Impact occurred due to horizontal movement
+								if (_delta.y > 0)
+									// Go to bottom of box
+									walkPos.y = objBounds.bottom + CLEAR_DIST_Y;
+								else if (_delta.y < 0)
+									// Go to top of box
+									walkPos.y = objBounds.top - CLEAR_DIST_Y;
+								else {
+									// Going straight horizontal, so choose best side
+									if (spritePt.y >= (objBounds.top + objBounds.height() / 2))
+										walkPos.y = objBounds.bottom + CLEAR_DIST_Y;
+									else
+										walkPos.y = objBounds.top - CLEAR_DIST_Y;
+								}
+
+								walkPos.x = (_delta.x >= 0) ? objBounds.left - CLEAR_DIST_X :
+									objBounds.right + CLEAR_DIST_X;
+							}
+
+							walkPos.x += people[AL]._imageFrame->_frame.w / 2;
+							people._walkDest = walkPos;
+							people._walkTo.push(walkPos);
+							people.setWalking();
+						}
+						break;
+
+					case DELTA:
+						_position.x += 200;
+						break;
+
+					default:
+						break;
+					}
+				}
+			}
+		}
+	}
+}
+
+/*----------------------------------------------------------------*/
+
+/**
+ * Load the data for the action
+ */
+void ActionType::load(Common::SeekableReadStream &s) {
+	char buffer[12];
+
+	_cAnimNum = s.readByte();
+	_cAnimSpeed = s.readByte();
+	if (_cAnimSpeed & 0x80)
+		_cAnimSpeed = -(_cAnimSpeed & 0x7f);
```

Comment:
The game uses the minus sign in many places to indicate reversement.. this includes horizontal flipping of images, reading flags to indicate flipping the result flag or whether to set or reset a flag. In just a few places such as this they were using a bit in the value. So I converted it to be negative numbers for consistency.